src/pyk/proof/reachability.py

```diff
@@ -296,21 +345,54 @@ def summary(self) -> Iterable[str]:
 
 class APRProver:
     proof: APRProof
+    kcfg_explore: KCFGExplore
     _is_terminal: Callable[[CTerm], bool] | None
     _extract_branches: Callable[[CTerm], Iterable[KInner]] | None
     _abstract_node: Callable[[CTerm], CTerm] | None
 
+    main_module_name: str
+    some_dependencies_module_name: str
+    all_dependencies_module_name: str
+
     def __init__(
         self,
         proof: APRProof,
+        kcfg_explore: KCFGExplore,
         is_terminal: Callable[[CTerm], bool] | None = None,
         extract_branches: Callable[[CTerm], Iterable[KInner]] | None = None,
         abstract_node: Callable[[CTerm], CTerm] | None = None,
     ) -> None:
         self.proof = proof
+        self.kcfg_explore = kcfg_explore
         self._is_terminal = is_terminal
         self._extract_branches = extract_branches
         self._abstract_node = abstract_node
+        self.main_module_name = self.kcfg_explore.kprint.definition.main_module_name
+
+        subproofs: list[Proof] = (
+            [Proof.read_proof(i, proof_dir=proof.proof_dir) for i in proof.subproof_ids]
+            if proof.proof_dir is not None
+            else []
+        )
+
+        apr_subproofs: list[APRProof] = [pf for pf in subproofs if type(pf) is APRProof]
```

Comment:
```suggestion
        apr_subproofs: list[APRProof] = [pf for pf in subproofs if isinstance(pf, APRProof)]
```

This will also include `APRBMCProof` types as well.